MusicPlayer/MusicPlayer/ViewController.swift

```diff
@@ -15,31 +17,136 @@ final class ViewController: UIViewController {
+    private var player : AVPlayer?
+    private var remoteCommandCenter = MPRemoteCommandCenter.shared()
+    private var playCommand: Any?
+    private var pauseCommand: Any?
+    
+    private let pauseRate: Double = 0
+    private let playRate: Double = 1
     
     override func viewDidLoad() {
         super.viewDidLoad()
         setNeedsStatusBarAppearanceUpdate()
+        initPlayer()
+        setupBackground()
         
         if let audio = selectedAudio {
-            thumbnailImageView.image = UIImage(named: audio.thumbnailSource)
-            titleLabel.text = audio.title
-            artistLabel.text = audio.artist
+            thumbnailImageView.image = UIImage(named: audio.thumbnailImageView)
+            titleLabel.text = audio.titleLabel
+            artistLabel.text = audio.artistLabel
         }
     }
     
+    override func viewDidDisappear(_ animated: Bool) {
+        removeCommandCenter()
+    }
+    
     override var preferredStatusBarStyle: UIStatusBarStyle {
         .lightContent
     }
     
     @IBAction private func playButtonPressed(_ sender: UIButton) {
-        var imageSystemName: String
-        isPlaying
-        ? (imageSystemName = "play.circle.fill")
-        : (imageSystemName = "pause.circle.fill")
-        isPlaying.toggle()
-        playButton.setImage(UIImage(systemName: imageSystemName), for: .normal)
+        isPlaying ? (
+            self.configureStatePause(),
+            MPNowPlayingInfoCenter.default().playbackState = .paused
+        ) : (
+            self.configureStatePlay(),
+            MPNowPlayingInfoCenter.default().playbackState = .playing
+        )
+        
+        addActionsToControlCenter()
+        updateInfoCenter()
+    }
+    
+    private func initPlayer() {
+        guard let url = Bundle.main.url(forResource: selectedAudio?.url, withExtension: "mp3")
+        else {
+            return
+        }
+        
+        player = AVPlayer(url: url)
+    }
+    
+    private func setupBackground() {
+        do {
+            try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default, options: [])
+            try AVAudioSession.sharedInstance().setActive(true)
+        } catch {
+            print("Error when setup background")
+        }
+    }
+    
+    private func removeCommandCenter(){
+        MPNowPlayingInfoCenter.default().nowPlayingInfo = nil
+        MPNowPlayingInfoCenter.default().playbackState = .stopped
+        remoteCommandCenter.playCommand.isEnabled = false
+        remoteCommandCenter.playCommand.removeTarget(playCommand)
+        remoteCommandCenter.pauseCommand.isEnabled = false
+        remoteCommandCenter.pauseCommand.removeTarget(pauseCommand)
+    }
+    
+    private func addActionsToControlCenter(){
+        remoteCommandCenter.playCommand.isEnabled = true
+        remoteCommandCenter.pauseCommand.isEnabled = true
+                        
+        playCommand = remoteCommandCenter.playCommand.addTarget {event in
```

Comment:
```suggestion
        playCommand = remoteCommandCenter.playCommand.addTarget { event in
```
Tương tự check kỹ nha em